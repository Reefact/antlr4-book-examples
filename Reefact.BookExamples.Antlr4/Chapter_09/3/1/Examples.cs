#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_09._3._1 {

    [UseReporter(typeof(BeyondCompareReporter), typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        [GraphicalTree("good_syntax.svg")]
        public void good_syntax() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("[1^2]");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("bad_syntax.svg")]
        public void bad_syntax() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("[]");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

    }

}