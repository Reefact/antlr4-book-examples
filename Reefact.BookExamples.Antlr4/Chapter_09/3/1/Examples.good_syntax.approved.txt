graph TD
	1["group"] --> 2["["]
	1 --> 3["expr"]
	3 --> 4["atom"]
	4 --> 5["1"]
	3 --> 6["^"]
	3 --> 7["2"]
	1 --> 8["]"]

classDef default fill:#fff,stroke:#000,stroke-width:0.25px;
classDef error color:#fff,fill:#FF0000,stroke:#000,stroke-width:0.25px;